src/calculateScore.js

```diff
@@ -0,0 +1,31 @@
+// calculateScore is a pure function. The score of a suggestion does not depend on other suggestions, which allows for simpler logic
+const calculateScore = (params, suggestion) => {
+    if (!params.latitude || !params.longitude) {
+        // If no position is specified, we'll first show the cities starting with the query, then those that countain the query
+        // Bigger cities will be displayed first, as they are more likely to be searched
+        const index = suggestion.name.indexOf(params.q); // Always a positive integer, otherwise the suggestion wouldnt have been returned by psql
+        return (index === 0 ? 0.5 : 0) + suggestion.population / 50000000;
+        // To ensure the score never exceeds 1, we could use the double of the most populated city instead of 50000000
+        // However, this quick solution will yield the same results and won't cause a score above 1 for long enough
+    } else {
+        const distance = Math.hypot(parseFloat(params.latitude) - suggestion.latitude, parseFloat(params.longitude) - suggestion.longitude);
+        if (distance === 0) return 1;
```

Comment:
This equality to zero check is not foolproof.
`Math.hypot` returns a number, including floating point number. JS Number follows IEEE 754 standard and so floating point representation can be imprecise. It's better to check for equality with something minuscule like `if (distance < 1e-20)`.